doc/sphinx/source/esmvalcore/preprocessor.inc

```diff
@@ -1,36 +1,215 @@
+land and 50m for ocean masks).
+
+Note that for masking out ice sheets, the preprocessor uses a different
+function, to ensure that both land and sea or ice can be masked out without
+losing generality. To mask ice out, `mask_landseaice` can be used:
+
+.. code-block:: yaml
+
+  preprocessors:
+    preproc_mask:
+      mask_landseaice:
+        mask_out: ice
+
+and requires only one argument:
+* mask_out: either landsea or ice.
+
+As in the case of `mask_landsea`, the preprocessor automatically retrieves the
+`fx: sftgif` mask.
+
+Another option is to just read the fx masks as any other CMOR variable and use
+it within a diagnostic script. This can be done in the variable dictionary by
+specifiying the desired fx variables (masks):
+
+.. warning::
+  Code snippet, text and link to function to be added (after #1037 and #1075
+  are closed).
+
+
+
+Horizontal regridding
+=====================
+.. warning::
+   Documentation of  _regrid.py (part 2) to be added
+
+
+
+Masking of missing values
+=========================
+.. warning::
+   Documentation of  _mask.py (part 2) to be added
+
+
+Multi-model statistics
+======================
+
+.. warning::
+   Documentation of _multimodel.py to be added.
+
+Information on maximum memory required: In the most general case, we can set upper limits on the maximum memory the anlysis will require:
+
+Ms = (R + N) x F_eff - F_eff - when no multimodel analysis is performed;
+Mm = (2R + N) x F_eff - 2F_eff - when multimodel analysis is performed;
+
+where
+
+Ms: maximum memory for non-multimodel module
+Mm: maximum memory for multimodel module
+R: computational efficiency of module (typically 2-3)
+N: number of datasets
+F_eff: average size of data per dataset where F_eff = e x f x F
+where e is the factor that describes how lazy the data is (e = 1 for fully realized data)
+and f describes how much the data was shrunk by the immediately previous module eg
+time extraction, area selection or level extraction; note that for fix_data f relates only to the time extraction, if data is exact in time (no time selection) f = 1 for fix_data
+
+so for cases when we deal with a lot of datasets (R + N = N), data is fully realized, assuming an average size of 1.5GB for 10 years of 3D netCDF data, N datasets will require
+
+
+Ms = 1.5 x (N - 1) GB
+Mm = 1.5 x (N - 2) GB
+
+
+Time operations
+===============
+
+The _time.py module contains a broad set of functions to subset data and apply
+statistical operators along the temporal coordinate of the input data:
+
+1. extract_time: extract a time range from a cube.
+2. extract_season: extract only the times that occur within a specific season.
+3. extract_month: extract only the times that occur within a specific month.
+4. time_average: take the weighted average over the time dimension.
+5. seasonal_mean: produce a mean for each season (DJF, MAM, JJA, SON)
+6. annual_mean: produce an annual or decadal mean.
```

Comment:
So the key "annual_mean" also produces a decadal mean? That is not very obvious...

Reply:
A decadal mean is an annual mean over 10 years :smiley:

Reply:
Yes, so probably instead of `decadal = true/false` we could think about more options.

Reply:
Sure. But what I actually just wanted to point out is that users might not find/suspect the decadal option under the key "annual_mean".

Reply:
How about we keep the `annual_mean` function, but add a generic tool `time_statistics`? This tool could follow the behaviour of `area_statistics` and `volume_statistics`.

We could then have for instance, the decadal median, or the maximum value over 5 year periods, or standard deviation of 25 year periods... etc...

Reply:
Can you open an issue for this? 
It is anyway beyond the scope of this PR.